src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -404,6 +404,23 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
                         o.EnableDependencyTracking = false;
                     });
                 }
+
+                if (SystemEnvironment.Instance.IsApplicationInsightsAgentEnabled())
+                {
+                    // Skip sending user generated logs to AI and QuickPulse if worker AI agent is configured, worker will send these logs to AI and Quickpulse service.
+                    builder.Services.Configure<LoggerFilterOptions>(options => options.AddFilter<ApplicationInsightsLoggerProvider>((category, logLevel) =>
+                    {
+                        // match Function.<FunctionName>.User
+                        if (category.Length > 14 && category.EndsWith(".User", StringComparison.Ordinal))
```

Comment:
I'm really curious about the perf of doing this vs lazily filling and checking a HashSet. This is fine for now as it won't impact any language but Java... and even then, a subset of scenarios, I believe. We can re-evaluate later if needed.

Reply:
You just snuck in before me with your benchmark comment! :-)